### ISSUE TITLE
Setup lock screen is asked when migrating to SDK version 9.2.0

### ISSUE THREAD

**Issue by mnaturel-niji** [forcedotcom/SalesforceMobileSDK-Android#2252]
1. Version of Mobile SDK Used: 9.2.0
2. Issue found in Native App or Hybrid App: A setup lock screen is displayed when migrating from SDK 9.1.1 to SDK 9.2.0
3. OS Version: Android 11
4. Device: Android Emulator Pixel 4
5. Steps to reproduce: Install an app using SDK below 9.2.0 (e.g 9.1.1), configure the connected app so that it does not require passcode after login, login into the app, update the app using SDK 9.2.0.
6. Actual behavior: The setup lock screen is shown after updating the app whereas the connected app is configured so that no passcode is required after login. The user can logout from this setup screen and login. After a new login, the setup screen is no more displayed but this extra step is a little confusing for users.
7. Expected Behavior: I would expect, since no passcode is required in the connected app, the user will not see this setup screen and stay logged in after updating the app.
8. Error Log: N/A

I have seen in the release notes what has been done in this version regarding passcode and I have inspected the code which is run after upgrading to SDK 9.2.0, especially the one in `SalesforceSDKUpgradeManager`. But I don't understand why in the method `upgradeTo9Dot2()` we have for global prefs: 
```
if (globalPrefs.contains(KEY_TIMEOUT) && globalPrefs.contains(KEY_PASSCODE_LENGTH)) {
    globalEditor.putBoolean(SCREEN_LOCK, true).apply();
}
```
and then for the user prefs:
```
if (orgPrefs.contains(KEY_TIMEOUT) &&  orgPrefs.contains(KEY_PASSCODE_LENGTH)) {
    userPrefs.edit().putBoolean(SCREEN_LOCK, true).apply();
}
```

Is the presence of KEY_TIMEOUT and KEY_PASSCODE_LENGTH in the prefs sufficient to say a passcode is required ? If so, I may have missed some configuration steps for the connected app.

I attached the screen which is displayed and logs of the global shared prefs just before upgrading.

![screenschot-lock-screen](https://user-images.githubusercontent.com/46314705/143906670-3670c697-01b2-4cd4-b359-9a9bc870a946.png)
![map-global-prefs](https://user-images.githubusercontent.com/46314705/143906697-ef89e5c4-9f3f-45fd-b68b-cabaeda196ff.png)

**Comment by brandonpage** [comment]
Hi @mnaturel-niji,  sorry of the very slow response.  

This is a very unusual scenario because the only way you could have the `KEY_TIMEOUT` and `KEY_PASSCODE_LENGTH` keys in prefs is if the Passcode feature is enabled in the connected app.  We retrieve this info at login and the system previous to 9.2.0 remained until the user had logged out of all orgs no matter what requirements the current org had.   We explicitly do not backup shared prefs and no data should remain through logout.  Is there something in your app that could be causing that to happen?

**Comment by mnaturel-niji** [comment]
Thanks for the response @brandonpage , I will try to investigate what configuration is in place with the Back End team. It may have not been well configured. But it remains strange since so far we didn't have any screen to configure such passcode. I will let you know.

**Comment by brandonpage** [comment]
@mnaturel-niji I have put a fix for this issue into Mobile SDK 9.2.1, which was just released.
